app/models/v2/report.rb

```diff
@@ -7,9 +7,43 @@ class Report < ApplicationRecord
     self.table_name = :reports
     self.primary_key = :id
 
+    SYSTEM_COUNT = lambda do
+      AN::NamedFunction.new(
+        'COUNT', [V2::System.arel_table[:id]]
+      ).filter(Pundit.policy_scope(User.current, V2::System).arel.ast.cores.first.wheres.first)
+    end
+
     # To prevent an autojoin with itself, there should not be an inverse relationship specified
     belongs_to :policy, class_name: 'V2::Policy', foreign_key: :id # rubocop:disable Rails/InverseOf
-    has_many :tailorings, class_name: 'V2::Tailoring', through: :policy
-    has_many :systems, class_name: 'V2::System', through: :tailorings
+    belongs_to :account
+
+    belongs_to :profile, class_name: 'V2::Profile'
+    has_one :security_guide, through: :profile, class_name: 'V2::SecurityGuide'
+    has_many :tailorings, class_name: 'V2::Tailoring', foreign_key: :policy_id
+    has_many :test_results, class_name: 'V2::TestResult', dependent: nil, through: :tailorings
+    has_many :systems, class_name: 'V2::System', through: :test_results
+
```

Comment:
You'd also need `has_many :policy_systems, foreign_key: :policy_id` for the `system_count` to know how many systems have never reported. Then you might need to tie it further to systems, but you probably have to name it differently than `:systems` as it is already taken. Maybe `:assigned_systems`?

Reply:
```
has_many :assigned_systems, class_name: 'V2::System', through: :policy_systems, source: :system
```